src/lib/Output/Generator/Xml.php

```diff
@@ -4,264 +4,129 @@
+        if (empty($attributes)) {
+            $jsonValue = $value;
+        } else {
+            $jsonValue = new Json\JsonObject($this->json);
+            foreach ($attributes as $attributeName => $attributeValue) {
+                $jsonValue->{'@' . $attributeName} = $attributeValue;
+            }
 
-        foreach ($attributes as $attributeName => $attributeValue) {
-            $this->xmlWriter->startAttribute($attributeName);
-            $this->xmlWriter->text($attributeValue);
-            $this->xmlWriter->endAttribute();
+            $jsonValue->{'#'} = $value;
         }
 
-        $this->xmlWriter->text((string)$value);
-    }
-
-    /**
-     * End value element.
-     *
-     * @param string $name
-     */
-    public function endValueElement($name)
-    {
-        $this->checkEndValueElement($name);
-
-        $this->xmlWriter->endElement();
-    }
-
-    /**
-     * Start list.
-     *
-     * @param string $name
-     */
-    public function startList($name)
-    {
-        $this->checkStartList($name);
+        if ($this->json instanceof Json\ArrayObject || $this->json instanceof Data\ArrayList) {
+            $this->json->append($jsonValue);
+        } else {
+            $this->json->$name = $jsonValue;
+        }
     }
 
-    /**
-     * End list.
-     *
-     * @param string $name
-     */
-    public function endList($name)
+    #[\Override]
+    public function endDocument(mixed $data): string
     {
-        $this->checkEndList($name);
-    }
+        parent::checkEndDocument($data);
 
-    /**
-     * Start attribute.
-     *
-     * @param string $name
-     * @param string $value
-     */
-    public function startAttribute($name, $value)
-    {
-        $this->checkStartAttribute($name);
+        $data = $this->getData();
 
-        $this->xmlWriter->startAttribute($name);
-        $this->xmlWriter->text((string)$value);
-    }
+        if (!$data instanceof Json\JsonObject) {
+            throw new \LogicException('Expected an instance of JsonObject');
+        }
 
-    /**
-     * End attribute.
-     *
-     * @param string $name
-     */
-    public function endAttribute($name)
-    {
-        $this->checkEndAttribute($name);
+        $vars = get_object_vars($data);
+        $encoderContext = $this->getEncoderContext($vars);
 
-        $this->xmlWriter->endAttribute();
-    }
+        $normalizers = [
+            new ArrayListNormalizer(),
+            new JsonObjectNormalizer(),
+            new ArrayObjectNormalizer(),
+            new ObjectNormalizer(),
+        ];
+        $encoders = [new XmlEncoder()];
+        $serializer = new Serializer($normalizers, $encoders);
 
-    /**
-     * Get media type.
-     *
-     * @param string $name
-     *
-     * @return string
-     */
-    public function getMediaType($name)
-    {
-        return $this->generateMediaTypeWithVendor($name, 'xml', $this->vendor);
+        return $serializer->serialize($data, 'xml', $encoderContext);
```

Comment:
CS: technically `$serializer` is redundant and can be inlined to:
```php
return (new Serializer($normalizers, $encoders))->serialize($data, 'xml', $encoderContext);
```

Reply:
Personally, I'd prefer it as is.

While you can inline it, it merges two operations (serializer instantiation and actual serialization) in a single line. It's a matter of preference. In this case the instantiation is rather complex (given arguments in preceeding variables) and it's clearly separated by whitespace that this is what we're doing.

We could make a private method instead.